jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/UriTemplatePathSpec.java

```diff
@@ -312,6 +323,20 @@ private boolean isValidBasicLiteralCodepoint(int codepoint)
         return false;
     }
 
+    @Override
+    public int compareTo(PathSpec other)
+    {
+        if (other instanceof UriTemplatePathSpec)
+        {
+            UriTemplatePathSpec otherUriPathSpec = (UriTemplatePathSpec)other;
+            return otherUriPathSpec.logicalDeclaration.compareTo(this.logicalDeclaration);
```

Comment:
Cool, so this works just because we always want to pick the logicalDeclaration that has the first 'e'. So its essentially doing the left to right matching described in the spec.

Will this still pick the right one if the logicalDeclarations are different length?

Reply:
The "Logical Declaration" isn't the internal "e" and "v" things from before.

It's taking the URI Template variables and making them "*", as pointed out here -> https://github.com/eclipse/jetty.project/pull/4788#issuecomment-615349675

The behavior of logical declarations of different lengths does not seem to be covered in tests, i'll add a few.

Reply:
New test added in commit 5397f505fc9f88145afbd16d6a1e1eaa229f2985